components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/client/authentication/OAuthClientAuthnService.java

```diff
@@ -60,6 +75,24 @@ public List<OAuthClientAuthenticator> getClientAuthenticators() {
     public OAuthClientAuthnContext authenticateClient(HttpServletRequest request, Map<String, List> bodyContentParams) {
 
         OAuthClientAuthnContext oAuthClientAuthnContext = new OAuthClientAuthnContext();
+        try {
+            if (Boolean.parseBoolean(isFapiApplication(extractClientId(request)))) {
+                if (!isMTLSEnforced(request)) {
+                    setErrorToContext(OAuth2ErrorCodes.INVALID_REQUEST, "Transport certificate not passed " +
+                            "through the request or the certificate is not valid", oAuthClientAuthnContext);
+                    return oAuthClientAuthnContext;
+                }
+                if (!isRegisteredClientAuthMethod(request, bodyContentParams)) {
+                    setErrorToContext(OAuth2ErrorCodes.INVALID_REQUEST, "Request does not follow the " +
+                            "registered token endpoint auth method", oAuthClientAuthnContext);
+                    return oAuthClientAuthnContext;
+                }
+            }
+        } catch (OAuthClientAuthnException e) {
+            setErrorToContext(OAuth2ErrorCodes.INVALID_REQUEST, "Error occurred while validating the " +
```

Comment:
will it be better to log the error here?

Reply:
Fixed in 1227a3c7eda7627ef2cf6a63f8e54714f18af3a9